Source/Core/Structures/RingBuffer.h

```diff
@@ -16,8 +16,17 @@ class RingBuffer
 	RingBuffer(const RingBuffer&) = delete;
 	RingBuffer(RingBuffer&& o) noexcept = delete;
 
-	RingBuffer& operator=(const RingBuffer&) = delete;
-	RingBuffer& operator=(RingBuffer&& other) noexcept = delete;
+	RingBuffer& operator=(const RingBuffer& other) = delete;
+	RingBuffer& operator=(RingBuffer&&) noexcept = delete;
```

Comment:
```suggestion
	RingBuffer& operator=(RingBuffer&&) noexcept = default;
```
This would default the implementation to the compiler generated function.